src/api/spec/controllers/webui/obs_factory/staging_projects_controller_spec.rb

```diff
@@ -0,0 +1,133 @@
+                                             review_state: 'new',
+                                             target_project: factory.name,
+                                             target_package: target_package.name,
+                                             source_project: source_package.project.name,
+                                             source_package: source_package.name))
+            end
+          end
+          let(:backend_response) do
+            <<-TEXT.strip_heredoc
+              613048: Needs to come in sync with Mesa changes (libwayland-egl1 is also built by Mesa.spec)
+              617649: Needs a perl fix - https://rt.perl.org/Public/Bug/Display.html?id=133295
+            TEXT
+          end
+
+          before do
+            allow_any_instance_of(Package).to receive(:file_exists?).with('ignored_requests').and_return(true)
+            stub_request(:get, backend_url).and_return(body: backend_response)
+
+            get :index, params: { project: factory }
+          end
+
+          it { expect(response).to have_http_status(:success) }
+          it { expect(response).to render_template(:index) }
+          it { expect(assigns(:backlog_requests_ignored)).to contain_exactly(create_review_requests.first) }
+          it { expect(assigns(:backlog_requests)).to contain_exactly(create_review_requests.last) }
+          it { expect(assigns(:requests_state_new)).to contain_exactly(create_review_requests_in_state_new.last) }
+          it { expect(assigns(:project)).to eq(factory) }
+        end
+
+        context 'without content' do
+          before do
+            allow_any_instance_of(Package).to receive(:file_exists?).with('ignored_requests').and_return(true)
+
+            stub_request(:get, backend_url).and_return(body: '')
+            get :index, params: { project: factory }
+          end
+
+          it { expect(response).to have_http_status(:success) }
+          it { expect(response).to render_template(:index) }
+          it { expect(assigns(:ignored_requests)).to be_falsey }
+          it { expect(assigns(:backlog_requests_ignored)).to be_empty }
+          it { expect(assigns(:backlog_requests)).to be_empty }
+          it { expect(assigns(:requests_state_new)).to be_empty }
+          it { expect(assigns(:project)).to eq(factory) }
+        end
+      end
+    end
+  end
+
+  describe 'GET #show' do
+    context 'with a existent factory_staging_project' do
+      subject { get :show, params: { project: factory, project_name: 'A' } }
+
+      it { expect(subject).to have_http_status(:success) }
+      it { expect(subject).to render_template(:show) }
+    end
+
+    context 'with a non-existent factory_staging_project' do
+      subject { get :show, params: { project: factory, project_name: 'B' } }
+
+      it { expect(subject).to have_http_status(:found) }
+      it { expect(subject).to redirect_to(root_path) }
+    end
+  end
+end
```

Comment:
In general this feels a bit like overkill to test `::Backend::Api::Sources::Package.file` and `if`/`select`. Do you really think we need to test this to this detail?

Reply:
Is to catch if we have the file, and if it also have any content or not... the only way to test it is by its content in if/select vars...

Reply:
Sure but we already test that `:Backend::Api::Sources::Package.file` works and there is no logic. At least we could not test the *"negative*" case when there is no file.

Reply:
Ok then